### PR TITLE
Add test for CASSANDRA-10215

### PR DESCRIPTION
**Pull request by beobal**
Only the coordinator should select which index, if any, should be used for a query. Previously, the index selection process would be run every time the index was accessed and on every replica.

### CODE REVIEW

File: secondary_indexes_test.py [hunk 1]
```diff
@@ -266,6 +266,70 @@ def wait_for_schema_agreement(self, session):
             self.wait_for_schema_agreement(session)
```
Comment (ptnapoleon) [inline secondary_indexes_test.py]: There should only be one new line between these two tests.

File: secondary_indexes_test.py [hunk 2]
```diff
@@ -266,6 +266,70 @@ def wait_for_schema_agreement(self, session):
...
+        # node2 is the coordinator for the query
+        check_trace_events(query.trace,
+                           "Index mean cardinalities are b_index:[0-9]*. Scanning with b_index.",
+                           [("127.0.0.1", 0, 0), ("127.0.0.2", 1, 1), ("127.0.0.3", 0, 0)])
+        # check that the index is used on each node, really we only care that the matching
+        # message appears on every node, so the max count is not important
+        check_trace_events(query.trace,
+                           "Executing read on ks.cf using index b_index",
+                           [("127.0.0.1", 1, 200), ("127.0.0.2", 1, 200), ("127.0.0.3", 1, 200)])
+
```
Comment (ptnapoleon) [inline secondary_indexes_test.py]: There should be two new lines before the start of a new class.

File: secondary_indexes_test.py [hunk 3]
```diff
@@ -266,6 +266,70 @@ def wait_for_schema_agreement(self, session):
...
+        """
+        Checks that the index to use is selected (once) on the coordinator and
+        included in the serialized command sent to the replicas.
+        @jira_ticket CASSANDRA-10215
+        """
+        cluster = self.cluster
+        cluster.populate(3).start()
+        node1, node2, node3 = cluster.nodelist()
+        conn = self.patient_cql_connection(node1)
+        session = conn
```
Comment (ptnapoleon) [inline secondary_indexes_test.py]: What is the point of defining conn? Why not `session = self.patient_cql_connection(node1)`?

**Comment by ptnapoleon** [comment]
+1 on the test logic. If you could make those whitespace changes to make Travis happy, that would be great.

File: secondary_indexes_test.py [hunk 4]
```diff
@@ -266,6 +266,70 @@ def wait_for_schema_agreement(self, session):
...
+        session.max_trace_wait = 120
+        session.execute("CREATE KEYSPACE ks WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': '1'};")
+        session.execute("CREATE TABLE ks.cf (a text PRIMARY KEY, b text);")
+        session.execute("CREATE INDEX b_index ON ks.cf (b);")
+        num_rows = 100
+        for i in range(num_rows):
+            indexed_value = i % (num_rows / 3)
+            # use the same indexed value three times
+            session.execute("INSERT INTO ks.cf (a, b) VALUES ('%d', '%d');" % (i, indexed_value))
```
Comment (mambocab) [inline secondary_indexes_test.py]: Could you please, here and elsewhere, use `.format` instead of `%` for string formatting? Thanks.

Comment (beobal) [inline secondary_indexes_test.py]: Sorry yes, bad habits die hard. I've fixed this instance & will try to remember in future.

File: secondary_indexes_test.py (hunk 3 above)
Comment (beobal) [inline secondary_indexes_test.py]: No clue what that was for, it was just a copypasta from `test_low_cardinality_indexes`. I've fixed it in both places.

**Comment by beobal** [comment]
squashed and merged

**Comment by mambocab** [comment]
:+1: thank you!

**Comment by ptnapoleon** [comment]
For easy searching, this was merged at 3929e9f2954aeb9e12d4bae623e175e1b388cc31

**Pull request closed without merging**